node/pkg/ethereum/getlogsimpl.go

```diff
@@ -0,0 +1,201 @@
+	e.BasePoller.SetLogger(l)
+}
+
+func (e *GetLogsImpl) DialContext(ctx context.Context, rawurl string) (err error) {
+	e.Query.poller = e.BasePoller
+	return e.BasePoller.DialContext(ctx, rawurl)
+}
+
+func (e *GetLogsImpl) NewAbiFilterer(address ethCommon.Address) (err error) {
+	return e.BasePoller.NewAbiFilterer(address)
+}
+
+func (e *GetLogsImpl) NewAbiCaller(address ethCommon.Address) (err error) {
+	return e.BasePoller.NewAbiCaller(address)
+}
+
+func (e *GetLogsImpl) GetCurrentGuardianSetIndex(ctx context.Context) (uint32, error) {
+	return e.BasePoller.GetCurrentGuardianSetIndex(ctx)
+}
+
+func (e *GetLogsImpl) GetGuardianSet(ctx context.Context, index uint32) (ethAbi.StructsGuardianSet, error) {
+	return e.BasePoller.GetGuardianSet(ctx, index)
+}
+
+type GetLogsPollSubscription struct {
+	errOnce   sync.Once
+	err       chan error
+	quit      chan error
+	unsubDone chan struct{}
+}
+
+var ErrUnsubscribedForGetLogs = errors.New("unsubscribed")
+
+func (sub *GetLogsPollSubscription) Err() <-chan error {
+	return sub.err
+}
+
+func (sub *GetLogsPollSubscription) Unsubscribe() {
+	sub.errOnce.Do(func() {
+		select {
+		case sub.quit <- ErrUnsubscribedForGetLogs:
+			<-sub.unsubDone
+		case <-sub.unsubDone:
+		}
+		close(sub.err)
+	})
+}
+
+func (e *GetLogsImpl) WatchLogMessagePublished(ctx, timeout context.Context, sink chan<- *ethAbi.AbiLogMessagePublished) (ethEvent.Subscription, error) {
+	e.Query.sink = sink
+
+	e.Query.sub = &GetLogsPollSubscription{
+		err: make(chan error, 1),
+	}
+
+	return e.Query.sub, nil
+}
+
+func (e *GetLogsImpl) TransactionReceipt(ctx context.Context, txHash ethCommon.Hash) (*ethTypes.Receipt, error) {
+	return e.BasePoller.TransactionReceipt(ctx, txHash)
+}
+
+func (e *GetLogsImpl) TimeOfBlockByHash(ctx context.Context, hash ethCommon.Hash) (uint64, error) {
+	return e.BasePoller.TimeOfBlockByHash(ctx, hash)
+}
+
+func (e *GetLogsImpl) ParseLogMessagePublished(log ethTypes.Log) (*ethAbi.AbiLogMessagePublished, error) {
+	return e.BasePoller.ParseLogMessagePublished(log)
+}
+
+func (e *GetLogsImpl) SubscribeForBlocks(ctx context.Context, sink chan<- *common.NewBlock) (ethereum.Subscription, error) {
+	return e.BasePoller.SubscribeForBlocks(ctx, sink)
+}
+
+type GetLogsQuery struct {
+	logger          *zap.Logger
+	networkName     string
+	ContractAddress ethCommon.Address
+	prevBlockNum    *big.Int
+	client          *ethClient.Client
+	poller          *PollImpl
+	sink            chan<- *ethAbi.AbiLogMessagePublished
+	sub             *GetLogsPollSubscription
+}
+
+func (f *GetLogsQuery) SetLogger(l *zap.Logger, netName string) {
+	f.logger = l
+	f.networkName = netName
+}
+
+func (f *GetLogsQuery) DialContext(ctx context.Context, rawurl string) (err error) {
+	timeout, cancel := context.WithTimeout(ctx, 15*time.Second)
+	defer cancel()
+	f.client, err = ethClient.DialContext(timeout, rawurl)
+
+	if err != nil {
+		return err
+	}
+
+	return err
```

Comment:
```suggestion
	return err
```

Either do it this way, or better, change the return to `return nil`. For such a short function, there isn't much of a benefit for named return args IMO.

Reply:
good comment, can pick up later :)